src/Storages/MergeTree/MutateTask.cpp

```diff
@@ -1461,6 +1483,8 @@ bool MutateTask::prepare()
         ctx->materialized_indices = ctx->interpreter->grabMaterializedIndices();
         ctx->materialized_projections = ctx->interpreter->grabMaterializedProjections();
         ctx->mutation_kind = ctx->interpreter->getMutationKind();
+        /// Always disable filtering in mutations, we want to read all rows
+        ctx->interpreter->setApplyDeletedMask(false);
```

Comment:
Maybe elaborate on this line. It's unclear how we will preserve previous delete masks if they existed. So let's clarify the comment.

Reply:
Wrote a more detailed comment